docs/examples/Makefile

```diff
@@ -0,0 +1,34 @@
+project_root_dir := $(strip $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST)))))
```

Comment:
Try to simplify this Makefile based on this Makefile: https://github.com/Ericsson/codechecker/blob/master/web/tests/projects/cpp/Makefile

Reply:
To start a make process  from other directory is allowed. This line prevents accidentally deletions in this case by clean target.

Reply:
This file is not subject of the howto document. It only works and nice.

Reply:
This is marked as resolved but it is not resolved yet. Please do not mark comment as resolved untill it is trully resolved or give me a reason why it is not resolved.

Reply:
I agree with @csordasmarton to simplify the makefile to the simple suggested one. At the end of the day it is only consists of 3 trivial files.

Reply:
- As you mentioned: "Later if someone wants to modify this file...". Why would somebody want to change the makefile? Because it changed the hierarchy of this "project". Your suggestion requires makefile modification too if some file is added or removed. Mine is resistant for that.

- In the howto document sometimes we clean the project and re-make it. Clean feature is not supported by your suggested solution.

Our purpose with that howto is to inspire users to play with it. There are some other issues in these C sources. If the user tries to repair them then he should re-make the project. But your suggested solution will not make if the user changes the divide.h file.

- The make can be called not only from that directory where the makefile resides. For example:
>cd <repo root> && make -f docs/examples/Makefile
should work. But your suggested makefile will not.

I think I have explained the requirements that were in my mind while creating that makefile.

I think if these requirements are valid then the "simplified" makefile will be at least comlex as mine is.

So, please specify your requirements against the makefile. And I will simplify it following them.

Reply:
With my solution there will be no generated files (output is redirected to /dev/null) so there is no need for clean target.

You don't have to step into the directory where the Makefile can be found, you can use the `-C` option of the make command for it if you would like to:
```sh
$ make -C docs/examples/
make: Entering directory '~/codechecker/docs/examples'
cc -Iincl -c src/divide.c -o /dev/null
cc -Iincl -c src/main.c -o /dev/null
make: Leaving directory '~/codechecker/docs/examples'
```

I still not recommend this complex Makefile for this simple example project but I will not block this PR. So on my side I finished the review of this patch.